### PR TITLE
setToDefault - use Objects.equal() to compare values

### PR DESCRIPTION
**Pull request by kthoms**

### CODE REVIEW

File: org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/preferences/FixedScopedPreferenceStore.java [hunk 1]
```diff
@@ -703,7 +705,7 @@ public void setToDefault(String name) {
 			// removing a non-existing preference is a no-op so call the Core
 			// API directly
 			getStorePreferences().remove(name);
-			if (oldValue != defaultValue){
+			if (!Objects.equal(oldValue, defaultValue)) {
```
Comment (cdietrich) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/preferences/FixedScopedPreferenceStore.java]: i wonder if this should be fixed in platform as well

Comment (kthoms) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/preferences/FixedScopedPreferenceStore.java]: where in platform?

Comment (cdietrich) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/preferences/FixedScopedPreferenceStore.java]: org.eclipse.ui.preferences.ScopedPreferenceStore.setToDefault(String)

Comment (kthoms) [inline org.eclipse.xtext.ui/src/org/eclipse/xtext/ui/editor/preferences/FixedScopedPreferenceStore.java]: I do think so.

**Pull request merged**